### PR TITLE
Save / sync chat text box messages per channel

### LINKED ISSUE #14423: Messages being written in lounge arent transferred to in-game textbox
They are saved/displayed once we get back out of the lobby. But when interrupted in stable/tourney play it's almost always possible to finish typing before the map starts

**video:** https://youtu.be/lKyCN7iaIK8
**osu!lazer version:** 2021.820.0-lazer

### PR DESCRIPTION
**Pull request by Joehuu**
- Resolves https://github.com/ppy/osu/issues/14423

| Chat Overlay | Multiplayer |
| --- | --- |
| osu!_joEdA5mPmU | osu!_7sD1y8qTUG |

### CODE REVIEW

File: osu.Game/Overlays/Chat/ChatTextBar.cs [hunk 1]
```diff
@@ -128,9 +128,8 @@ protected override void LoadComplete()
                 chattingTextContainer.FadeTo(showSearch ? 0 : 1);
                 searchIconContainer.FadeTo(showSearch ? 1 : 0);
 
-                // Clear search terms if any exist when switching back to chat mode
-                if (!showSearch)
-                    OnSearchTermsChanged?.Invoke(string.Empty);
+                if (showSearch)
+                    OnSearchTermsChanged?.Invoke(chatTextBox.Current.Value);
```
Comment (bdach) [inline osu.Game/Overlays/Chat/ChatTextBar.cs:132]: Explanation for this change? Doesn't seem related to what this PR is claiming to be doing.

Comment (Joehuu) [reply osu.Game/Overlays/Chat/ChatTextBar.cs:132]: I just wanted the behavior on search to be consistent even though it isn't an actual channel. If it's meant to be cleared every time, I'll revert.

Comment (bdach) [reply osu.Game/Overlays/Chat/ChatTextBar.cs:132]: I don't know if it should or shouldn't clear but these kinds of weird flourishes being put into mostly unrelated PRs without prior discussion or explanation are tiring me out if I'm honest. I guess it's fine?

**Review by bdach — APPROVED** [review]
